tests/devices/unit_tests/test_gridscan.py

```diff
@@ -25,97 +27,49 @@ def discard_status(st: Status | DeviceStatus):
 
 
 @pytest.fixture
-def fast_grid_scan(request):
-    FakeFastGridScan = make_fake_device(FastGridScan)
-    fast_grid_scan: FastGridScan = FakeFastGridScan(
-        name=f"test fake FGS: {request.node.name}"
-    )
-    fast_grid_scan.scan_invalid.pvname = ""
-    yield fast_grid_scan
+async def zebra_fast_grid_scan(request):
+    async with DeviceCollector(mock=True):
+        zebra_fast_grid_scan = ZebraFastGridScan(name="fake_FGS", prefix="FGS")
 
+    return zebra_fast_grid_scan
 
-def test_given_settings_valid_when_kickoff_then_run_started(
-    fast_grid_scan: FastGridScan,
-):
-    when(fast_grid_scan.scan_invalid).get().thenReturn(False)
-    when(fast_grid_scan.position_counter).get().thenReturn(0)
-
-    mock_run_set_status = mock()
-    when(fast_grid_scan.run_cmd).put(ANY).thenReturn(mock_run_set_status)
-    fast_grid_scan.status.subscribe = lambda func, **_: func(1)  # type: ignore
 
-    status = fast_grid_scan.kickoff()
-
-    status.wait()
-    assert status.exception() is None
+@pytest.fixture
+async def panda_fast_grid_scan(request):
```

Comment:
Should: Why do we need `request` here and in the other fixture? They don't seem to be used